### PR TITLE
ENG-10045, make DRConsumerDrIdTracker seerializable, remove dead code.

### PR DESCRIPTION
**Pull request by luyangco**
Change-Id: I640707164625fb1b8d490ace1dc558073d6d6075

### CODE REVIEW

File: src/frontend/org/voltdb/DRConsumerDrIdTracker.java [hunk 1]
```diff
@@ -99,7 +105,12 @@ public DRConsumerDrIdTracker(JSONObject jsObj) throws JSONException {
         for (int ii = 0; ii < drIdRanges.length(); ii++) {
             JSONObject obj = drIdRanges.getJSONObject(ii);
             String startDrIdStr = obj.keys().next();
-            m_map.add(range(Long.valueOf(startDrIdStr), obj.getLong(startDrIdStr)));
+            long start = Long.valueOf(startDrIdStr);
+            long end = obj.getLong(startDrIdStr);
+            if (start > end) {
+                throw new IllegalArgumentException(start + " after " + end);
```
Comment (nshi) [inline src/frontend/org/voltdb/DRConsumerDrIdTracker.java]: I think the `Range` static constructor in `range()` does the validation already.

File: src/frontend/org/voltdb/DRConsumerDrIdTracker.java [hunk 2]
```diff
@@ -17,22 +17,28 @@
...
 import com.google_voltpatches.common.collect.Range;
 import com.google_voltpatches.common.collect.RangeSet;
 import com.google_voltpatches.common.collect.TreeRangeSet;
-import org.json_voltpatches.JSONArray;
-import org.json_voltpatches.JSONException;
-import org.json_voltpatches.JSONObject;
-import org.voltdb.iv2.UniqueIdGenerator;
 
-public class DRConsumerDrIdTracker {
+public class DRConsumerDrIdTracker implements Serializable {
```
Comment (nshi) [inline src/frontend/org/voltdb/DRConsumerDrIdTracker.java]: Where do we use this as a serializable object? I'm just concerned of adding a third serialization to this class.

Comment (luyangco) [inline src/frontend/org/voltdb/DRConsumerDrIdTracker.java]: The third serialization has been used in https://github.com/VoltDB/pro/blob/HeterogeneousDR-IntegrationBranch/src/frontend/org/voltdb/sysprocs/SnapshotRestore.java#L1291

File: src/frontend/org/voltdb/DRConsumerDrIdTracker.java (hunk 1 above)
Comment (luyangco) [inline src/frontend/org/voltdb/DRConsumerDrIdTracker.java]: I'll remove the redundant validations.

File: src/frontend/org/voltdb/iv2/Site.java [hunk 3]
```diff
@@ -485,6 +485,11 @@ public void recoverWithDrAppliedTrackers(Map<Integer, Map<Integer, DRConsumerDrI
         }
 
         @Override
+        public void resetDrAppliedTracker() {
+            m_maxSeenDrLogsBySrcPartition.clear();
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/Site.java]: Also reset the last applied spUniqueId and mpUniqueId.

Comment (luyangco) [inline src/frontend/org/voltdb/iv2/Site.java]: Done.

**Comment by luyangco** [comment]
Squash merged externally.

**Pull request closed without merging**